app/packages/looker/src/lookers/abstract.ts

```diff
@@ -191,7 +191,24 @@ export abstract class AbstractLooker<
     return this.sampleOverlays;
   }
 
-  protected dispatchEvent(eventType: string, detail: any): void {
+  getSizeBytesEstimate() {
+    let size = 1;
+    if (this.state.dimensions && !this.state.error) {
+      const [w, h] = this.state.dimensions;
+      size += w * h * 4;
+    }
+
+    if (!this.sampleOverlays?.length) {
+      return size;
+    }
+
+    for (let index = 0; index < this.sampleOverlays.length; index++) {
+      size += this.sampleOverlays[index].getSizeBytes();
+    }
```

Comment:
For overlays, we might want to add one more case in `sizer` for bitmaps since they have the largest memory footprint. Something along the lines of:

```
if (typeof object === "object" && object.constructor.name === "ImageBitmap") {
    size += object.height * object.width * 4;
}
```